packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+
+In LoopBack Next topics, our focus is these three core concepts: controllers, components, and sequence.
+
+Alright, let's get the foundation techniques under the belt.
+
+
+
+# Foundation
+
+Here, I'm going to summarize a few key concepts and a list of tools I've found very useful.  With that, I got a big picture without getting bogged down to the details of each technology; and, at the same time, learned a way to go back the the details when needed.
+
+LoopBack Next is built on TypeScript and OpenAPI.  TypeScript is build on ECMAScript 6.  Quickly browsing those technologies from the ground up help solidify our steps. 
+
+### [ECMAScript 6](https://en.wikipedia.org/wiki/ECMAScript#6th_Edition_-_ECMAScript_2015)
+
+1. ECMAScript 6 — New Features: Overview & Comparison -- http://es6-features.org/#Constants 
+2. The spec -- https://www.ecma-international.org/ecma-262/6.0/ECMA-262.pdf
+3. Arrow Functions -- http://es6-features.org/#ExpressionBodies
+
+The overview & comparison site is very helpful to me.  Reading through the short list of code comparison between ES5 and ES6 gave me a good perspective.  I frequently go back to the site.  When I wanted to understand what's not there, I went to the spec and other options, but that rarely occurred.
+
+### [TypeScript](https://www.typescriptlang.org/) : Type System on ECMAScript 6
+
+1. Getting Started With TypeScript -- https://basarat.gitbooks.io/typescript/docs/getting-started.html 
+2. Declaration Files -- https://basarat.gitbooks.io/typescript/docs/types/ambient/d.ts.html
+3. Decorators -- https://www.typescriptlang.org/docs/handbook/decorators.html
+
+When you need to `npm install async` don't forget to `npm install -S "@types/async"
+
+### [OpenAPI](https://www.openapis.org/)
+
+1. https://editor.swagger.io
+
+Note (LoopBack Next specific): In a LoopBack Next app, typically you don't put the header portion of the api spec(see below) because LoopBack Next adds it when internalizing it.  The swagger editor will complain without the header portion.
+```
+{
+  "swagger": "2.0",
+  "basePath": "/",
+  "info": {
+    "title": "LoopBack Application",
+    "version": "1.0.0",
+  },
+  paths:{
+      ...
+  }
+}
+```
+Also note that LoopBack Next provides `validateApiSpec` in the testlab so that you can validate it in `npm test` of your app.  Isn't that cool?
+
+
+
+# LoopBack Next
+
+We're going to examine three key concepts: controller & apiSpec, components & providers, and custom sequence.  Working sample codes are available in several places, one of which is [Hello World](https://github.com/strongloop/loopback-next-hello-world).  I'm going to focus on specific conceptual roadblocks I stumbled on.
+
+Skeleton of the client code looks like this:
+```
```

Comment:
Please specify the language as here - add `js` or `ts` right after the three backticks. Similarly in other code examples below.